tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,524 @@
+            "one",                         // testname
+            false,                         // is_force
+            false,                         // is_replace
+            ["valid1"].to_vec(),           // assert_keys
+        ) {
+            Ok(()) => panic!("unexpected test success"),
+            Err(err) => {
+                // Error would be like "keys with name 'one' already exist".
+                assert!(format!("{}", err).contains("keys with name"));
+                assert!(format!("{}", err).contains("already exist"));
+            }
+        }
+
+        // "update-ssh-keys --no-replace --add-force one valid2.pub"
+        _ = add_one_ssh_key(
+            ctx.ssh_dir.to_str().unwrap(), // ssh_dir
+            "valid2",                      // pubkeyname
+            "one",                         // testname
+            true,                          // is_force
+            false,                         // is_replace
+            ["valid1"].to_vec(),           // assert_keys
+        );
+    }
+
+    #[test]
+    fn test_add_two() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --add one valid1.pub"
+        test_add_one_file();
+
+        // "update-ssh-keys --add two valid2.pub"
+        _ = add_one_ssh_key(
+            ctx.ssh_dir.to_str().unwrap(), // ssh_dir
+            "valid2",                      // pubkeyname
+            "two",                         // testname
+            false,                         // is_force
+            true,                          // is_replace
+            ["valid1", "valid2"].to_vec(), // assert_keys
+        );
+    }
+
+    #[test]
+    fn test_del_one() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --add one valid1.pub"
+        test_add_one_file();
+
+        // "update-ssh-keys --delete one valid1.pub"
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid1", "one");
+    }
+
+    #[test]
+    fn test_del_two() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --delete one valid1.pub"
+        // "update-ssh-keys --delete two valid2.pub"
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid1", "one");
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid2", "two");
+
+        // "update-ssh-keys --add one valid1.pub"
+        // "update-ssh-keys --add two valid2.pub"
+        test_add_two();
+
+        // "update-ssh-keys --delete two valid2.pub"
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid2", "two");
+    }
+
+    #[test]
+    fn test_disable() {
+        let ctx = setup_tests();
+
+        // "update-ssh-keys --delete one valid1.pub"
+        // "update-ssh-keys --delete two valid2.pub"
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid1", "one");
+        del_one_ssh_key(ctx.ssh_dir.to_str().unwrap(), "valid2", "two");
+
+        // "update-ssh-keys --add one valid1.pub"
+        // "update-ssh-keys --add two valid2.pub"
+        test_add_two();
```

Comment:
That's wrong because `test_add_two` operates on its own private tmpdir, you should pass `ctx`.

Reply:
Done